src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,139 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = new Node<T>(null, value, null);
+            tail = head;
+
+        } else {
+            tail.next = new Node<>(tail, value, null);
+            tail = tail.next;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> nodeToInsert = new Node<>(null, value, null);
+        if (index == 0) {
+            nodeToInsert.next = head;
+            head.prev = nodeToInsert;
+            head = nodeToInsert;
+            size++;
+            return;
+        }
+        Node<T> current = getNodeByIndex(index);
+        nodeToInsert.prev = current.prev;
+        nodeToInsert.next = current;
+        current.prev.next = nodeToInsert;
+        current.prev = nodeToInsert;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null) {
+            return;
+        }
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> nodeByIndex = getNodeByIndex(index);
+        return nodeByIndex.element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> nodeByIndex = getNodeByIndex(index);
+        T previousValue = nodeByIndex.element;
+        nodeByIndex.element = value;
+        return previousValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> current = getNodeByIndex(index);
+        T element = current.element;
+        unlink(current);
+        return element;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = head;
+        for (int i = 0; i < size; i++) {
+            if (object == current.element || object != null && object.equals(current.element)) {
+                unlink(getNodeByIndex(i));
+                return true;
+            }
+            current = current.next;
+        }
         return false;
     }
 
+    private void unlink(Node<T> node) {
```

Comment:
you don't need to check the index for validity because you do it in the getNodeByIndex() method